sphinxcontrib_hdl_diagrams/__init__.py

```diff
@@ -384,16 +410,25 @@ def render_diagram(self, code, options, format, skin, yosys_script):
     relfn = posixpath.join(self.builder.imgpath, fname)
     outfn = path.join(self.builder.outdir, self.builder.imagedir, fname)
 
+    yosys = self.builder.config.hdl_diagram_yosys
+
     if source_ext == '.py':
         module = 'top'
         ilfn = path.join(self.builder.outdir, self.builder.imagedir, options['outname'] + '.il')
         nmigen_to_rtlil(source_path, ilfn)
         source_path = ilfn
     elif source_ext == '.il' or source_ext == '.v':
         module = options['module']
+    elif source_ext == '.vhd' or source_ext == '.vhdl':
+        if yosys == "yowasp":
+            raise HDLDiagramError("Cannot use YoWASP for VHDL (yet)")
+        module = options['module']
+        ilfn = path.join(self.builder.outdir, self.builder.imagedir, options['outname'] + '.v')
+        vhdl_to_verilog(source_path, ilfn, module, self.builder.config.hdl_diagram_ghdl, yosys)
```

Comment:
I'm not sure to understand this approach. My perception is that you are supporting Verilog sources only or VHDL sources only. Then, you convert VHDL to Verilog as a preprocess step, and then use exactly the same functions you'd use for "regular" Verilog sources. Therefore, generating drawings of mixed-language designs is not supported by this PR. Is this correct? If so, I would explain it somewhere, together with inviting people to contribute for making mixed-language diagramas possible.

Reply:
You are 100% right. After going back through the fomu workshop mixed-hdl example, I want to implement this differently than what I have now. I'm going to continue this discussion back on #65.